src/varname.jl

```diff
@@ -747,3 +748,62 @@ function vsym(expr::Expr)
         error("Malformed variable name `$(expr)`!")
     end
 end
+
+"""
+    index_to_str(i)
+
+Generates a string representation of the index `i`, or a tuple thereof.
+"""
+index_to_str(i::Integer) = string(i)
+index_to_str(r::UnitRange) = "$(first(r)):$(last(r))"
+index_to_str(::Colon) = ":"
+index_to_str(s::ConcretizedSlice{T,R}) where {T,R} = "ConcretizedSlice(" * repr(s.range) * ")"
+index_to_str(t::Tuple) = "(" * join(map(index_to_str, t), ", ") * ",)"
+
+"""
+    optic_to_nt(optic)
+
+Convert an optic to a named tuple representation.
+"""
+optic_to_nt(::typeof(identity)) = (type = "identity",)
+optic_to_nt(::PropertyLens{sym}) where {sym} = (type = "property", field = String(sym))
+optic_to_nt(i::IndexLens) = (type = "index", indices = index_to_str(i.indices))
+optic_to_nt(c::ComposedOptic) = (type = "composed", outer = optic_to_nt(c.outer), inner = optic_to_nt(c.inner))
+
+
+"""
+    nt_to_optic(nt)
+
+Convert a named tuple representation back to an optic.
+"""
+function nt_to_optic(nt)
+    if nt.type == "identity"
+        return identity
+    elseif nt.type == "index"
+        return IndexLens(eval(Meta.parse(nt.indices)))
+    elseif nt.type == "property"
+        return PropertyLens{Symbol(nt.field)}()
+    elseif nt.type == "composed"
+        return nt_to_optic(nt.outer) ∘ nt_to_optic(nt.inner)
+    end
+end
+
+"""
+    vn_to_string(vn::VarName)
+
+Convert a `VarName` as a string (via an intermediate named tuple).
+"""
+vn_to_string(vn::VarName) = repr((sym = String(getsym(vn)), optic = optic_to_nt(getoptic(vn))))
+
+"""
+    vn_from_string(str)
+
+Convert a string representation of a `VarName` back to a `VarName`.
+
+NOTE: This function should only be used with trusted input, as it uses `eval`
+and `Meta.parse` to parse the string.
+"""
+function vn_from_string(str)
+    new_fields = eval(Meta.parse(str))
+    return VarName{Symbol(new_fields.sym)}(nt_to_optic(new_fields.optic))
+end
```

Comment:
I don't entirely like this, but the only reasonable alternative is to handwrite a parser and manually allow a curated list of expressions to be `eval`'d.

If our main use case is for serialising Chains and other Turing.jl objects to disk, then I think it is reasonable to expect the user to make sure their own input is trusted.

Reply:
I'm wary of pulling in a serialisation library – firstly, it's an extra dep for this library, and secondly I'm not sure how it will work with the lenses that we are trying to serialise – do we have to also define serialisation methods for those types (which means either type piracy on our end, or including that functionality in Accessors itself, which is an extra dep for them)? I could be wrong on that point, but that was my superficial understanding from a brief look.